framework/include/serializer.hpp

```diff
@@ -71,7 +80,21 @@ namespace gvk {
 			mArchive(aMode == serializer::mode::serialize ?
 				std::variant<deserialize, serialize>{ serializer::serialize(aCacheFilePath) } :
 				std::variant<deserialize, serialize>{ serializer::deserialize(aCacheFilePath) })
-		{}
+		{
+			std::uint32_t version = SERIALIZER_CACHE_FILE_VERSION;
+			if (mode() == serializer::mode::serialize)
+			{
+				std::get<serialize>(mArchive)(std::forward<std::uint32_t>(version));
```

Comment:
```suggestion
				std::get<serialize>(mArchive)(version);
```
Why the `std::forward`? 
`version` should just be passed as an ordinary reference, shouldn't it?!

Reply:
True... but I just realised, that the constructor can just call `archive` to avoid code duplication because it does the same thing.